tests/time_tests/src/timetests_helper/timer.cpp

```diff
@@ -14,16 +14,19 @@ using time_point = std::chrono::high_resolution_clock::time_point;
 
 namespace TimeTest {
 
+
 Timer::Timer(const std::string &timer_name) {
   name = timer_name;
   start_time = std::chrono::high_resolution_clock::now();
+  StatisticsWriter::Instance().addOrderCount(name);
 }
 
 Timer::~Timer() {
   float duration = std::chrono::duration_cast<std::chrono::microseconds>(
                        std::chrono::high_resolution_clock::now() - start_time)
                        .count();
-  StatisticsWriter::Instance().write({name, duration});
+  StatisticsWriter::Instance().addToTimeStructure({name, duration});
+  StatisticsWriter::Instance().deleteOrderCount();
```

Comment:
let's rename `deleteOrderCount()` to `deleteTimer(name, duration)`, so `addToTimeStructure()` can be incapsulated in `StatisticsWriter` and called directly in `deleteTimer`